Document Satellite Ansible Content Collections (#228)
theforeman/foreman-documentation: guides/common/modules/proc_listing-using-satellite-ansible-modules.adoc

[id="listing-using-satellite-ansible-modules_{context}"]
== Listing the {Project} Ansible Modules

> You can list the installed {Project} Ansible modules by navigating to the following directory:

Author: thanks. prefer a dedicated step/command for clarity than just in a sentence.

ifeval::["{build}" == "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/redhat/satellite/plugins/modules/
----

> Alternatively, you can also see the complete list of {Project} Ansible modules and other related information at https://cloud.redhat.com/ansible/automation-hub/redhat/satellite/docs by logging into the Automation Hub.

Reviewer: Please write the same sentence for the upstream in the below ifeval, but point to this Ansible Galaxy link: https://galaxy.ansible.com/theforeman/foreman.

endif::[]

ifeval::["{build}" != "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/modules/
----
endif::[]

[NOTE]
====
At the time of writing, the `ansible-doc -l` command does not list collections yet.
====

> All modules are in the `redhat.satellite` namespace and can be referred to as follows:

Reviewer: This needs to be rewritten twice with ifevals upstream because `redhat.satellite` won't work there.

Reviewer 2: Or just template that as `{ansible-collection-name}`?

Author: Ack, done.

ifeval::["{build}" == "satellite"]
----

> $ ansible-doc redhat.satellite.[module_name]

Reviewer: ```suggestion
$ ansible-doc redhat.satellite._module_name_
```
Add italics about square brackets for replaceable values. This is how we usually mark them up in docs. Note that for the italics to render, you need to add +quotes options above the command block.

Author: Added underscore. i've used + above command block in procedures to render the step number sequence. not sure what you mean here? + above command block rendered as is for me in this case.

----
endif::[]

ifeval::["{build}" != "satellite"]
----
$ theforeman.foreman._module_name
----
endif::[]

[NOTE]
====
All modules require the `server_url`, `username`, and `password` parameters.

> ====

Reviewer: This should not be a note really. Just a regular sentence somewhere where it makes sense. Ansible tells users if it needs more parameters anyway.

Author: Ack

[id="listing-using-satellite-ansible-modules_{context}"]

> == Listing the {Project} Ansible Modules

Reviewer: I suggest renaming this to *Viewing the {Project} Ansible Modules*. Listing sounds very narrow. What do you think @suyogsainkar ?

Author: Good suggestion, done.

You can view the installed {Project} Ansible modules by listing the content of the following directory:

ifeval::["{build}" == "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/redhat/satellite/plugins/modules/
----

Alternatively, you can also see the complete list of {Project} Ansible modules and other related information at https://cloud.redhat.com/ansible/automation-hub/redhat/satellite/docs in the Automation Hub.

endif::[]

ifeval::["{build}" != "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/modules/
----

Alternatively, you can also see the complete list of {Project} Ansible modules and other related information at https://galaxy.ansible.com/theforeman/foreman in the Ansible Galaxy.

endif::[]

[NOTE]
====
At the time of writing, the `ansible-doc -l` command does not list collections yet.
====

ifeval::["{build}" == "satellite"]

> All modules are in the `redhat.satellite` namespace and can be referred to as follows:

Reviewer: ```suggestion
All modules are in the `redhat.satellite` namespace and can be referred to in the format `redhat.satellite._module_name_`. For example, to display information on the `activation_key` module, enter the following command:
```
Add a little more context and implement the example

Author: Done

----

> $ ansible-doc redhat.satellite._module_name_

Reviewer: ```suggestion
$ ansible-doc redhat.satellite.activation_key
```

Author: Done

----
endif::[]

ifeval::["{build}" != "satellite"]

> All modules are in the `theforeman.foreman` namespace and can be referred to as follows:

Reviewer: ```suggestion
All modules are in the `theforeman.foreman` namespace and can be referred to in the format `theforeman.foreman._module_name_`. For example, to display information on the `katello_activation_key` module, enter the following command:
```

Author: Done

----

> $ ansible-doc theforeman.foreman._module_name

Reviewer: ```suggestion
$ ansible-doc theforeman.foreman.katello_activation_key
```

Author: Done

----
endif::[]

> Note that all modules require the `server_url`, `username`, and `password` parameters.

Reviewer: This bit is explained in the documentation for each separate module. This sentence might become untrue in future, which makes it hard to maintain. I expect Ansible users to be able to find the required parameters for modules they use. I suggest removing this sentennce completely.

Author: Done

[id="listing-using-satellite-ansible-modules_{context}"]
== Listing the {Project} Ansible Modules

You can list the installed {Project} Ansible modules by navigating to the following directory:

ifeval::["{build}" == "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/redhat/satellite/plugins/modules/
----
endif::[]

ifeval::["{build}" != "satellite"]
----
# ls /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/modules/
----
endif::[]

> Alternatively, you can also see the complete list of {Project} Ansible modules and other related information at https://cloud.redhat.com/ansible/automation-hub/redhat/satellite/docs in the Automation Hub.

Reviewer: Upstream, link to the upstream Ansible Galay modules page: https://galaxy.ansible.com/theforeman/foreman

Author: Ack.
